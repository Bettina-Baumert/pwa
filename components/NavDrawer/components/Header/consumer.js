import { consume } from 'redux-props';
import { historyPush } from '@shopgate/pwa-common/actions/router';
import { LOGIN_PATH } from '@shopgate/pwa-common/constants/RoutePaths';
import {
  getUserDisplayName,
  getUserEmail,
} from '@shopgate/pwa-common/selectors/user';

/**
 * @returns {Object}
 */
const mapProps = ({ dispatch, state }) => ({
  email: getUserEmail(state),
  name: getUserDisplayName(state),
  openLogin: () => dispatch(historyPush({
    pathname: LOGIN_PATH,
  })),
});

export default consume({
  mapProps,
});
