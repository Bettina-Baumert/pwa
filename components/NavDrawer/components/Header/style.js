/**
 * Copyright (c) 2017, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import cxs from 'cxs';
import colors from 'Styles/colors';
import variables from 'Styles/variables';

const container = cxs({
  marginBottom: 4,
});

const loggedIn = cxs({
  background: colors.accent,
  color: colors.light,
  padding: `${variables.gap.small + 1}px ${variables.gap.big}px ${variables.gap.small - 1}px`,
});

const ellipsis = {
  textOverflow: 'ellipsis',
  overflow: 'hidden',
  lineHeight: 1.3,
};

const welcome = cxs({
  fontWeight: 500,
  ...ellipsis,
});

const mail = cxs({
  ...ellipsis,
});

export default {
  container,
  loggedIn,
  welcome,
  mail,
};
