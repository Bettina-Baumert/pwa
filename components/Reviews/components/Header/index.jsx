import React from 'react';
import PropTypes from 'prop-types';
import { RATING_SCALE_DIVISOR } from 'Components/RatingStars/constants';
import I18n from '@shopgate/pwa-common/components/I18n';
import * as portals from '@shopgate/pwa-common-commerce/product/constants/Portals';
import Portal from '@shopgate/pwa-common/components/Portal';
import AverageRating from './components/AverageRating';
import WriteReviewLink from './components/WriteReviewLink';
import styles from './style';

/**
 * The header of the reviews component
 * @param {Object} rating The rating values
 * @param {bool} withTopGap Adds additional top gap when true.
 * @returns {JSX}
 */
const Header = ({ rating, withTopGap }) => {
  if (!rating) {
    return null;
  }
  const { average = 0 } = rating;
  const containerClass = withTopGap ? styles.withTopGapContainer : styles.container;

  if (!average) {
    return (
      <div className={styles.container}>
        <AverageRating rating={rating} />

          <div className={styles.noReviews}>
            <I18n.Text string="reviews.no_reviews" />
            <Portal name={portals.PRODUCT_WRITE_REVIEW_BEFORE} />
            <Portal name={portals.PRODUCT_WRITE_REVIEW}>
              <WriteReviewLink />
            </Portal>
            <Portal name={portals.PRODUCT_WRITE_REVIEW_AFTER} />
          </div>
        <Portal name={portals.PRODUCT_WRITE_REVIEW_BEFORE} />
      </div>
    );
  }

  return (
    <div className={containerClass} id="reviewsExcerpt">
      <AverageRating rating={rating} />
      <div className={styles.reviewsLine}>
        <I18n.Text string="reviews.rating" className={styles.averageRatingText}>
          <span className={styles.averageRatingNumber}>
            {average / RATING_SCALE_DIVISOR}
          </span>
        </I18n.Text>
        <Portal name={portals.PRODUCT_WRITE_REVIEW_BEFORE} />
        <Portal name={portals.PRODUCT_WRITE_REVIEW}>
          <WriteReviewLink />
        </Portal>
        <Portal name={portals.PRODUCT_WRITE_REVIEW_AFTER} />
      </div>
    </div>
  );
};

Header.propTypes = {
  rating: PropTypes.shape(),
  withTopGap: PropTypes.bool,
};

Header.defaultProps = {
  rating: null,
  withTopGap: false,
};

export default Header;
