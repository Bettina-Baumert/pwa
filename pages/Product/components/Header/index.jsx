/**
 * Copyright (c) 2017, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React from 'react';
import PropTypes from 'prop-types';
import AddToCartButton from './components/AddToCartButton';
import Rating from './components/Rating';
import Name from './components/Name';
import styles from './style';

/**
 * The product header component that displays textual information
 * - manufacturer
 * - shipping
 * - discount
 * - price
 * - price info
 * If not available or animating it will display placeholders
 * @param {Object} props The component props.
 * @returns {JSX}
 */
const ProductHeader = (props) => {
  // If we don't have data yet OR if the view is currently animating show placeholder only
  const placeholderProductReady = (!props.isAnimating && (props.product !== null));

  /**
   * Required to display placeholders.
   * If we don't have the product we don't know if it has discount, manufacturer, etc.
   * this object will decide what placeholders are visible
   * @type {Object}
   */
  const defaultData = {
    // White space because empty string is false. Needed to show placeholder.
    manufacturer: ' ',
    price: {},
    availability: {
      // White space because empty string is false. Needed to show placeholder.
      text: ' ',
    },
  };
  const product = (placeholderProductReady && props.product) ? props.product : defaultData;

  /**
   * Identical logic for shipping, like for the product. The data is requested with a separate
   * pipeline request, so it's placeholder needs separate handling.
   * @type {boolean}
   */
  const placeholderShippingReady = (!props.isAnimating && (props.shipping !== null));

  return (
    <div className={styles.content}>
      <AddToCartButton />
      <Rating />
      <Name />
    </div>
  );
};

ProductHeader.propTypes = {
  product: PropTypes.shape(),
  shipping: PropTypes.shape({
    currency: PropTypes.string,
    price: PropTypes.number,
  }),
};

ProductHeader.defaultProps = {
  product: null,
  shipping: {},
};

export default ProductHeader;
