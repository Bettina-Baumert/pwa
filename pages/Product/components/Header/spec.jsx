import React from 'react';
import { mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import mockRenderOptions from '@shopgate/pwa-common/helpers/mocks/mockRenderOptions';
import { defaultContext } from './../../__mocks__/context';
import { basicProductState } from './../mock';

const Mock = function () {
  return (<div />);
};
const mock = Mock;
const mockedStore = configureStore([thunk]);

jest.mock('Components/Reviews/components/Header', () => mock);
jest.mock('./../../context');

describe('<ProductHeader>', () => {
  const createComponent = (state) => {
    // eslint-disable-next-line global-require
    const ProductHeader = require('./index').default;
    const store = mockedStore(state);
    return mount(
      <Provider store={store}>
        <ProductHeader store={store} productId={state.product.currentProduct.productId} />
      </Provider>,
      mockRenderOptions
    );
  };

  const components = [
    'CTAButtons',
    'Manufacturer',
    'Availability',
    'Rating',
    'Name',
    'Price',
    'PriceStriked',
    'Shipping',
    'Tiers',
  ];

  it('should render', () => {
    const { productId } = basicProductState.product.currentProduct;
    defaultContext.productId = productId;
    const cmp = createComponent(basicProductState);
    expect(cmp).toMatchSnapshot();
    components.forEach((c) => {
      expect(cmp.find(c).exists()).toBe(true);
    });
  });
});
