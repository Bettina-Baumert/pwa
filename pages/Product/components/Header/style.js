/**
 * Copyright (c) 2017, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */
import { css } from 'glamor';
import colors from 'Styles/colors';
import variables from 'Styles/variables';

const content = css({
  position: 'relative',
  backgroundColor: colors.light,
  padding: variables.gap.big,
  borderTop: 'rgba(0, 0, 0, 0.05) 2px solid',
}).toString();

const manufacturer = css({
  fontSize: '1.0rem',
}).toString();

const priceInfo = css({
  fontSize: '0.625rem',
}).toString();

const infoContainer = css({
  width: '50%',
  alignSelf: 'flex-end',
}).toString();

const priceContainer = css({
  width: '50%',
  textAlign: 'right',
  alignSelf: 'flex-end',
}).toString();

const basePriceContainer = css({
  width: '50%',
  textAlign: 'right',
  alignSelf: 'flex-end',
  marginLeft: '50%',
}).toString();

const placeholder = {
  info: css({
    height: 16,
    width: '70%',
    marginTop: 5,
    marginBottom: 2,
  }).toString(),
  price: css({
    height: 20,
    width: '50px',
    display: 'inline-block',
  }).toString(),
};

export default {
  content,
  name,
  manufacturer,
  priceInfo,
  infoContainer,
  priceContainer,
  basePriceContainer,
  placeholder,
};
