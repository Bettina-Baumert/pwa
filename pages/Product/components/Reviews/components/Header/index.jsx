/**
 * Copyright (c) 2017, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React from 'react';
import PropTypes from 'prop-types';
import I18n from '@shopgate/pwa-common/components/I18n';
import AverageRating from './components/AverageRating';
import styles from './style';

/**
 * The header of the reviews component
 * @param {Object} rating The rating values
 * @returns {JSX}
 */
const Header = ({ rating }) => {
  let average = 0;

  if (rating && rating.count && rating.average) {
    average = rating.average;
  }

  if (!average) {
    return (
      <div className={styles.container}>
        <AverageRating rating={rating} />
        <div className={styles.noReviews}>
          <I18n.Text string="reviews.no_reviews" />
        </div>
      </div>
    );
  }

  return (
    <div className={styles.container}>
      <AverageRating rating={rating} />
      <div className={styles.reviewsLine}>
        <I18n.Text string="reviews.rating">
          <span className={styles.averageRating}>{average * 5}</span>
        </I18n.Text>
      </div>
    </div>
  );
};

Header.propTypes = {
  rating: PropTypes.shape(),
};

Header.defaultProps = {
  rating: null,
};

export default Header;
