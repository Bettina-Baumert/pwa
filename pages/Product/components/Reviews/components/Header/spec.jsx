/**
 * Copyright (c) 2017, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React from 'react';
import { mount } from 'enzyme';
import Header from './index';

describe('<Header />', () => {
  let header = null;
  const ratings = [
    {
      average: null,
      count: null,
      reviewCount: null,
    },
    {
      average: 0.7,
      count: 3,
      reviewCount: 3,
    },
  ];

  beforeEach(() => {
    header = mount(<Header rating={null} />);
  });

  it('should render empty', () => {
    expect(header.find('Header').exists()).toBe(true);
    header.setProps({ rating: ratings[0] });

    expect(header).toMatchSnapshot();
    expect(header.find('RatingStars').prop('value')).toEqual(0);
    expect(header.find('RatingCount').prop('count')).toEqual(0);
  });

  it('should render rating summary', () => {
    expect(header.find('Header').exists()).toBe(true);
    header.setProps({ rating: ratings[1] });

    expect(header).toMatchSnapshot();
    expect(header.find('RatingStars').prop('value')).toEqual(ratings[1].average);
    expect(header.find('RatingCount').prop('count')).toEqual(ratings[1].reviewCount);
  });
});
