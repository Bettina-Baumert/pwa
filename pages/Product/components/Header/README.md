# ProductHeader
---

Displays information related to the product.
All labels contain placeholders that are shown when data is either not there yet
or the view isAnimating.

> **Dependencies:**
<br> `<Manufacturer />`
<br> `<DiscountBadge />`
<br> `<Price />`
<br> `<StrikedPrice />`
<br> `<Shipping />`
<br> `<PlaceholderLabel />`

> **Dependents:** <br> `<ProductTemplate />`

## Getting Started

```
import { ProductHeader } from 'Templates/components';

<ProductHeader product={...} isAnimating />
```

## Props

### isAnimating (required)

_Type_: `boolean`  

Should be true when parent is animating, this will cause placeholders to be displayed

###### Usage:

```
<ProductHeader isAnimating={false} />
```

### product

_Default_: `null`  
_Type_: `Object`  

The BasicProduct object or `null` when product data is not available yet which will cause placeholders to be rendered.
> **Note:** When `isAnimating` is `true`, `product` will be ignored until animation is done!

###### Usage:

```
<ProductHeader product={...} isAnimating={false} />
```
