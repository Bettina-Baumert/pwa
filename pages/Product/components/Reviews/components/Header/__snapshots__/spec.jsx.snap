// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render empty 1`] = `
<Header
  rating={
    Object {
      "average": null,
      "count": null,
      "reviewCount": null,
    }
  }
>
  <div
    className="css-ud99sa"
  >
    <AverageRating
      rating={
        Object {
          "average": null,
          "count": null,
          "reviewCount": null,
        }
      }
    >
      <div
        className="css-t193cd"
      >
        <RatingStars
          className=""
          display="large"
          value={0}
        >
          <div
            className="css-13azwyo "
          >
            <div
              className="css-4s85e9"
            >
              <div
                className=" css-hvpjjr"
              >
                <Star
                  className=""
                  color={null}
                  size="2em"
                  viewBox="0 0 24 24"
                >
                  <Icon
                    className=""
                    color={null}
                    content="<path d=\\"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\\"/>"
                    size="2em"
                    viewBox="0 0 24 24"
                  >
                    <svg
                      className="css-yys9hb "
                      dangerouslySetInnerHTML={
                        Object {
                          "__html": "<path d=\\"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\\"/>",
                        }
                      }
                      style={
                        Object {
                          "fill": null,
                          "fontSize": "2em",
                        }
                      }
                      viewBox="0 0 24 24"
                      xmlns="http://www.w3.org/2000/svg"
                    />
                  </Icon>
                </Star>
              </div>
              <div
                className=" css-hvpjjr"
              >
                <Star
                  className=""
                  color={null}
                  size="2em"
                  viewBox="0 0 24 24"
                >
                  <Icon
                    className=""
                    color={null}
                    content="<path d=\\"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\\"/>"
                    size="2em"
                    viewBox="0 0 24 24"
                  >
                    <svg
                      className="css-yys9hb "
                      dangerouslySetInnerHTML={
                        Object {
                          "__html": "<path d=\\"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\\"/>",
                        }
                      }
                      style={
                        Object {
                          "fill": null,
                          "fontSize": "2em",
                        }
                      }
                      viewBox="0 0 24 24"
                      xmlns="http://www.w3.org/2000/svg"
                    />
                  </Icon>
                </Star>
              </div>
              <div
                className=" css-hvpjjr"
              >
                <Star
                  className=""
                  color={null}
                  size="2em"
                  viewBox="0 0 24 24"
                >
                  <Icon
                    className=""
                    color={null}
                    content="<path d=\\"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\\"/>"
                    size="2em"
                    viewBox="0 0 24 24"
                  >
                    <svg
                      className="css-yys9hb "
                      dangerouslySetInnerHTML={
                        Object {
                          "__html": "<path d=\\"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\\"/>",
                        }
                      }
                      style={
                        Object {
                          "fill": null,
                          "fontSize": "2em",
                        }
                      }
                      viewBox="0 0 24 24"
                      xmlns="http://www.w3.org/2000/svg"
                    />
                  </Icon>
                </Star>
              </div>
              <div
                className=" css-hvpjjr"
              >
                <Star
                  className=""
                  color={null}
                  size="2em"
                  viewBox="0 0 24 24"
                >
                  <Icon
                    className=""
                    color={null}
                    content="<path d=\\"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\\"/>"
                    size="2em"
                    viewBox="0 0 24 24"
                  >
                    <svg
                      className="css-yys9hb "
                      dangerouslySetInnerHTML={
                        Object {
                          "__html": "<path d=\\"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\\"/>",
                        }
                      }
                      style={
                        Object {
                          "fill": null,
                          "fontSize": "2em",
                        }
                      }
                      viewBox="0 0 24 24"
                      xmlns="http://www.w3.org/2000/svg"
                    />
                  </Icon>
                </Star>
              </div>
              <div
                className=" css-hvpjjr"
              >
                <Star
                  className=""
                  color={null}
                  size="2em"
                  viewBox="0 0 24 24"
                >
                  <Icon
                    className=""
                    color={null}
                    content="<path d=\\"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\\"/>"
                    size="2em"
                    viewBox="0 0 24 24"
                  >
                    <svg
                      className="css-yys9hb "
                      dangerouslySetInnerHTML={
                        Object {
                          "__html": "<path d=\\"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\\"/>",
                        }
                      }
                      style={
                        Object {
                          "fill": null,
                          "fontSize": "2em",
                        }
                      }
                      viewBox="0 0 24 24"
                      xmlns="http://www.w3.org/2000/svg"
                    />
                  </Icon>
                </Star>
              </div>
            </div>
            <div
              className="css-16y9zin"
            />
          </div>
        </RatingStars>
        <RatingCount
          count={0}
        />
      </div>
    </AverageRating>
    <div
      className="css-s8ghv1"
    >
      <Translate
        className=""
        params={Object {}}
        string="reviews.no_reviews"
      >
        <span
          className=""
        >
          reviews.no_reviews
        </span>
      </Translate>
    </div>
  </div>
</Header>
`;

exports[`<Header /> should render rating summary 1`] = `
<Header
  rating={
    Object {
      "average": 0.7,
      "count": 3,
      "reviewCount": 3,
    }
  }
>
  <div
    className="css-ud99sa"
  >
    <AverageRating
      rating={
        Object {
          "average": 0.7,
          "count": 3,
          "reviewCount": 3,
        }
      }
    >
      <div
        className="css-t193cd"
      >
        <RatingStars
          className=""
          display="large"
          value={0.7}
        >
          <div
            className="css-13azwyo "
          >
            <div
              className="css-4s85e9"
            >
              <div
                className=" css-hvpjjr"
              >
                <Star
                  className=""
                  color={null}
                  size="2em"
                  viewBox="0 0 24 24"
                >
                  <Icon
                    className=""
                    color={null}
                    content="<path d=\\"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\\"/>"
                    size="2em"
                    viewBox="0 0 24 24"
                  >
                    <svg
                      className="css-yys9hb "
                      dangerouslySetInnerHTML={
                        Object {
                          "__html": "<path d=\\"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\\"/>",
                        }
                      }
                      style={
                        Object {
                          "fill": null,
                          "fontSize": "2em",
                        }
                      }
                      viewBox="0 0 24 24"
                      xmlns="http://www.w3.org/2000/svg"
                    />
                  </Icon>
                </Star>
              </div>
              <div
                className=" css-hvpjjr"
              >
                <Star
                  className=""
                  color={null}
                  size="2em"
                  viewBox="0 0 24 24"
                >
                  <Icon
                    className=""
                    color={null}
                    content="<path d=\\"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\\"/>"
                    size="2em"
                    viewBox="0 0 24 24"
                  >
                    <svg
                      className="css-yys9hb "
                      dangerouslySetInnerHTML={
                        Object {
                          "__html": "<path d=\\"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\\"/>",
                        }
                      }
                      style={
                        Object {
                          "fill": null,
                          "fontSize": "2em",
                        }
                      }
                      viewBox="0 0 24 24"
                      xmlns="http://www.w3.org/2000/svg"
                    />
                  </Icon>
                </Star>
              </div>
              <div
                className=" css-hvpjjr"
              >
                <Star
                  className=""
                  color={null}
                  size="2em"
                  viewBox="0 0 24 24"
                >
                  <Icon
                    className=""
                    color={null}
                    content="<path d=\\"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\\"/>"
                    size="2em"
                    viewBox="0 0 24 24"
                  >
                    <svg
                      className="css-yys9hb "
                      dangerouslySetInnerHTML={
                        Object {
                          "__html": "<path d=\\"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\\"/>",
                        }
                      }
                      style={
                        Object {
                          "fill": null,
                          "fontSize": "2em",
                        }
                      }
                      viewBox="0 0 24 24"
                      xmlns="http://www.w3.org/2000/svg"
                    />
                  </Icon>
                </Star>
              </div>
              <div
                className=" css-hvpjjr"
              >
                <Star
                  className=""
                  color={null}
                  size="2em"
                  viewBox="0 0 24 24"
                >
                  <Icon
                    className=""
                    color={null}
                    content="<path d=\\"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\\"/>"
                    size="2em"
                    viewBox="0 0 24 24"
                  >
                    <svg
                      className="css-yys9hb "
                      dangerouslySetInnerHTML={
                        Object {
                          "__html": "<path d=\\"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\\"/>",
                        }
                      }
                      style={
                        Object {
                          "fill": null,
                          "fontSize": "2em",
                        }
                      }
                      viewBox="0 0 24 24"
                      xmlns="http://www.w3.org/2000/svg"
                    />
                  </Icon>
                </Star>
              </div>
              <div
                className=" css-hvpjjr"
              >
                <Star
                  className=""
                  color={null}
                  size="2em"
                  viewBox="0 0 24 24"
                >
                  <Icon
                    className=""
                    color={null}
                    content="<path d=\\"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\\"/>"
                    size="2em"
                    viewBox="0 0 24 24"
                  >
                    <svg
                      className="css-yys9hb "
                      dangerouslySetInnerHTML={
                        Object {
                          "__html": "<path d=\\"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\\"/>",
                        }
                      }
                      style={
                        Object {
                          "fill": null,
                          "fontSize": "2em",
                        }
                      }
                      viewBox="0 0 24 24"
                      xmlns="http://www.w3.org/2000/svg"
                    />
                  </Icon>
                </Star>
              </div>
            </div>
            <div
              className="css-16y9zin"
            >
              <div
                className=" css-hvpjjr"
              >
                <Star
                  className=""
                  color={null}
                  size="2em"
                  viewBox="0 0 24 24"
                >
                  <Icon
                    className=""
                    color={null}
                    content="<path d=\\"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\\"/>"
                    size="2em"
                    viewBox="0 0 24 24"
                  >
                    <svg
                      className="css-yys9hb "
                      dangerouslySetInnerHTML={
                        Object {
                          "__html": "<path d=\\"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\\"/>",
                        }
                      }
                      style={
                        Object {
                          "fill": null,
                          "fontSize": "2em",
                        }
                      }
                      viewBox="0 0 24 24"
                      xmlns="http://www.w3.org/2000/svg"
                    />
                  </Icon>
                </Star>
              </div>
              <div
                className=" css-hvpjjr"
              >
                <Star
                  className=""
                  color={null}
                  size="2em"
                  viewBox="0 0 24 24"
                >
                  <Icon
                    className=""
                    color={null}
                    content="<path d=\\"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\\"/>"
                    size="2em"
                    viewBox="0 0 24 24"
                  >
                    <svg
                      className="css-yys9hb "
                      dangerouslySetInnerHTML={
                        Object {
                          "__html": "<path d=\\"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\\"/>",
                        }
                      }
                      style={
                        Object {
                          "fill": null,
                          "fontSize": "2em",
                        }
                      }
                      viewBox="0 0 24 24"
                      xmlns="http://www.w3.org/2000/svg"
                    />
                  </Icon>
                </Star>
              </div>
              <div
                className=" css-hvpjjr"
              >
                <Star
                  className=""
                  color={null}
                  size="2em"
                  viewBox="0 0 24 24"
                >
                  <Icon
                    className=""
                    color={null}
                    content="<path d=\\"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\\"/>"
                    size="2em"
                    viewBox="0 0 24 24"
                  >
                    <svg
                      className="css-yys9hb "
                      dangerouslySetInnerHTML={
                        Object {
                          "__html": "<path d=\\"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\\"/>",
                        }
                      }
                      style={
                        Object {
                          "fill": null,
                          "fontSize": "2em",
                        }
                      }
                      viewBox="0 0 24 24"
                      xmlns="http://www.w3.org/2000/svg"
                    />
                  </Icon>
                </Star>
              </div>
              <div
                className=" css-hvpjjr"
              >
                <StarHalfIcon
                  className=""
                  color={null}
                  size="2em"
                  viewBox="0 0 24 24"
                >
                  <Icon
                    className=""
                    color={null}
                    content="<defs><path id=\\"a\\" d=\\"M0 0h24v24H0z\\"/></defs>
                    <clipPath id=\\"b\\"><use xlink:href=\\"#a\\" overflow=\\"visible\\"/></clipPath>
                    <path clip-path=\\"ul(#b)\\" d=\\"M12 2L9.2 8.6 2 9.2 7.5 14l-1.6 7 6.2-3.7m-.1-1.9V6.1m2.8 7.3\\"/>"
                    size="2em"
                    viewBox="0 0 24 24"
                  >
                    <svg
                      className="css-yys9hb "
                      dangerouslySetInnerHTML={
                        Object {
                          "__html": "<defs><path id=\\"a\\" d=\\"M0 0h24v24H0z\\"/></defs>
                          <clipPath id=\\"b\\"><use xlink:href=\\"#a\\" overflow=\\"visible\\"/></clipPath>
                          <path clip-path=\\"ul(#b)\\" d=\\"M12 2L9.2 8.6 2 9.2 7.5 14l-1.6 7 6.2-3.7m-.1-1.9V6.1m2.8 7.3\\"/>",
                        }
                      }
                      style={
                        Object {
                          "fill": null,
                          "fontSize": "2em",
                        }
                      }
                      viewBox="0 0 24 24"
                      xmlns="http://www.w3.org/2000/svg"
                    />
                  </Icon>
                </StarHalfIcon>
              </div>
            </div>
          </div>
        </RatingStars>
        <RatingCount
          count={3}
        >
          <Translate
            className="css-9pmgny"
            params={
              Object {
                "count": 3,
              }
            }
            string="reviews.review_count"
          >
            <span
              className="css-9pmgny"
            >
              reviews.review_count
            </span>
          </Translate>
        </RatingCount>
      </div>
    </AverageRating>
    <div
      className="css-132heyi"
    >
      <Translate
        className=""
        params={Object {}}
        string="reviews.rating"
      >
        <span
          className=""
        >
          reviews.rating
        </span>
      </Translate>
    </div>
  </div>
</Header>
`;
